main.go

```diff
@@ -140,21 +168,20 @@ func main() {
 	level.Info(logger).Log("msg", "Starting smartctl_exporter", "version", version.Info())
 	level.Info(logger).Log("msg", "Build context", "build_context", version.BuildContext())
 
-	var devices []string
+	var devices []Device
+	devices = scanDevices(logger)
+	level.Info(logger).Log("msg", "Number of devices found", "count", len(devices))
 	if len(*smartctlDevices) > 0 {
-		devices = *smartctlDevices
-	} else {
-		level.Info(logger).Log("msg", "No devices specified, trying to load them automatically")
-		devices = scanDevices(logger)
-		level.Info(logger).Log("msg", "Number of devices found", "count", len(devices))
+		level.Info(logger).Log("msg", "Devices specified", "devices", strings.Join(*smartctlDevices, ", "))
+		devices = filterDevices(devices, *smartctlDevices)
```

Comment:
This breaks for devices that are NOT detected by `smartctl --scan`, e.g. `--smartctl.device=/dev/nvme0n1` (because --scan returns `/dev/nvme0`).

Reply:
nvme0 - device
nvme0n1 - namespace 1 on device - not device!

Reply:
Agreed, namespaces (whose utility vs partitions are still unclear to me) are not what this exporter should be acting on. `smartctl`(passthrough) will return the same info IIRC for a device and any namespace.  While I could see where maybe the capacity of a namespace would vary, IMHO that’s beyond the scope.